File: Ref/Web/ReferenceDataUpdateService.Web/src/models/IRefCusCodeListAttributeName.ts
import { IEntity } from "./IEntity";

interface IRefCusCodeListAttributeName extends IEntity {
	ZXE_PK: string,
	ZXE_Name: string,
	ZXE_Description: string,
	ZXE_ZZK_NKCodeType: string,
	ZXE_ZZZ_NKDataGrouping: string,
	ZXE_IsMandatory: boolean,
	ZXE_AllowDuplicates: boolean,
	ZXE_IsValueMandatory: boolean,
	ZXE_ZZK_NKCodeTypeForValueList: string,
	ZXE_ValueDataType: string,
	ZXE_MinLengthOrValue: number,
	ZXE_MaxLengthOrValue: number,
	ZXE_DecimalPlaces: number,
	ZXE_ColumnCaption: string,
	ZXE_IsDateRangeUsed: boolean
}

export default IRefCusCodeListAttributeName;

File: Ref/Tools/DataSetUpdaterScriptGenerator/StorageData/IRefCusCodeListAttributeName.cs
using System;
using System.Collections.Generic;
using CargoWise.RefDbRepo.Client.Common;

namespace CargoWise.RefDbRepo.DataSetUpdaterScriptGenerator
{
	public partial interface IRefCusCodeListAttributeName : IDataSetStorage
	{
		Guid ZXE_PK { get; set; }
		string ZXE_Name { get; set; }
		string ZXE_Description { get; set; }
		string ZXE_ZZK_NKCodeType { get; set; }
		string ZXE_ZZZ_NKDataGrouping { get; set; }
		bool ZXE_IsMandatory { get; set; }
		bool ZXE_AllowDuplicates { get; set; }
		bool ZXE_IsValueMandatory { get; set; }
		string ZXE_ZZK_NKCodeTypeForValueList { get; set; }
		string ZXE_ValueDataType { get; set; }
		short ZXE_MinLengthOrValue { get; set; }
		decimal ZXE_MaxLengthOrValue { get; set; }
		byte ZXE_DecimalPlaces { get; set; }
		string ZXE_ColumnCaption { get; set; }
		bool ZXE_IsDateRangeUsed { get; set; }

		IEnumerable<IRefCusCodeListAttributeNameLanguage> RefCusCodeListAttributeNameLanguage { get; }
	}
}
